app/src/main/java/org/oppia/android/app/player/exploration/ExplorationActivityPresenter.kt

```diff
@@ -195,6 +211,27 @@ class ExplorationActivityPresenter @Inject constructor(
     ) as HintsAndSolutionExplorationManagerFragment?
   }
 
+  fun deleteCurrentProgressAndStopExploration() {
+    explorationDataController.deleteExplorationProgressById(
+      ProfileId.newBuilder().setInternalId(internalProfileId).build(),
+      explorationId
+    )
+    stopExploration()
+  }
+
+  fun deleteOldestSavedProgressAndStopExploration() {
+    // If the value of oldestExplorationId is null, it means that there was an error while
+    // retrieving the oldest saved checkpoint details. In this case, the exploration is exited
+    // without deleting the any checkpoints.
+    if (oldestExplorationId != null) {
+      explorationDataController.deleteExplorationProgressById(
+        ProfileId.newBuilder().setInternalId(internalProfileId).build(),
+        oldestExplorationId!!
```

Comment:
Why do we need to use `!!` complier should handle this since we have already added a null check

Reply:
`oldestExplorationId` is can be null but `deleteExplorationProgressById` expects it to be not null.

Reply:
`!!` removing this gives an error.

Reply:
Removed !!